### PR TITLE
[Cognitive Services] Support Identity, Encryption, UserOwnedStorage

### PR DESCRIPTION
**Pull request by yangyuan**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by yangyuan** [comment]
Some CI failed but seems due to server side issue

File: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/examples/CreateAccountWithEncryption.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{
+  "parameters": {
+    "api-version": "2017-04-18",
+    "subscriptionId": "f9b96b36-1f5e-4021-8959-51527e26e6d3",
+    "resourceGroupName": "felixwa-01",
+    "accountName": "testCreate1",
+    "account": {
+      "location": "West US",
+      "kind": "Emotion",
```
Comment (xiaomin-ms) [inline specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/examples/CreateAccountWithEncryption.json]: Emotion Kind has been decommissioned.

**Review by xiaomin-ms — APPROVED** [review]

File: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/examples/CreateAccountWithEncryption.json (hunk 1 above)
Comment (yangyuan) [reply specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/examples/CreateAccountWithEncryption.json]: this is swagger example, more for demonstrating/swagger validation purpose. Should be okay to use Emotion as sample kind.

**Review by mmyyrroonn — APPROVED** [review]
LGTM

File: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/examples/CreateAccountWithEncryption.json (hunk 1 above)
Comment (MyronFanQiu) [reply specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/examples/CreateAccountWithEncryption.json]: better to use valid kind. example files would be used to generate test/help or test the server. This should be correct.

Comment (MyronFanQiu) [reply specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/examples/CreateAccountWithEncryption.json]: not only for swagger validation purpose.

File: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/cognitiveservices.json [hunk 2]
```diff
@@ -1434,6 +1449,105 @@
...
+        },
+        "clientId": {
+          "type": "string",
+          "description": "Client App Id associated with this identity."
+        }
+      }
+    },
+    "Encryption": {
+      "properties": {
+        "keyVaultProperties": {
```
Comment (ryansbenson) [inline specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/cognitiveservices.json]: can 'keyVaultProperties' be used with 'keySource' set to 'Microsoft.CognitiveServices'?

File: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/cognitiveservices.json [hunk 3]
```diff
@@ -870,6 +874,17 @@
           "$ref": "#/definitions/NetworkRuleSet",
           "description": "A collection of rules governing the accessibility from specific network locations."
         },
+        "encryption": {
```
Comment (ryansbenson) [inline specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/cognitiveservices.json]: This constitutes a breaking change and requires a new API version.

See 'New property added to response'
https://microsoft.sharepoint.com/:w:/r/teams/azureresourcemanagerteam/_layouts/15/guestaccess.aspx?e=xEcppo&share=EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw

Comment (yangyuan) [reply specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/cognitiveservices.json]: I randomly opened a few recent PRs from other teams, they all adding properties to a response without creating a new version.
https://github.com/Azure/azure-rest-api-specs/pull/8498/files
https://github.com/Azure/azure-rest-api-specs/pull/8508/files
https://github.com/Azure/azure-rest-api-specs/pull/8468/files

The GET-PUT story doesn't apply in here, as our SDK update operation will ignore the missing properties and consider it as "no change".

File: specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/cognitiveservices.json (hunk 2 above)
Comment (yangyuan) [reply specification/cognitiveservices/resource-manager/Microsoft.CognitiveServices/stable/2017-04-18/cognitiveservices.json]: `keyVaultProperties` is not used for `Microsoft.CognitiveServices` source.

**Pull request merged**